internal/webdavmount/webdavmount.go

```diff
@@ -110,27 +110,30 @@ type webdavDir struct {
 var symlinksAreUnsupportedLogged = new(int32)
 
 func (d *webdavDir) Readdir(n int) ([]os.FileInfo, error) {
-	entries, err := d.entry.Readdir(d.ctx)
-	if err != nil {
-		return nil, errors.Wrap(err, "error reading directory")
-	}
+	var fis []os.FileInfo
 
-	if n > 0 && n < len(entries) {
-		entries = entries[0:n]
-	}
+	foundEntries := 0
 
-	var fis []os.FileInfo
+	err := d.entry.IterateEntries(d.ctx, func(innerCtx context.Context, e fs.Entry) error {
+		if n > 0 && n <= foundEntries {
+			return nil
```

Comment:
nit: returning nil does not stop iteration, perhaps return some marker error `errStop` and treat it as success.